src/eos-updater-fetch.c

```diff
@@ -75,29 +75,62 @@ update_progress (OstreeAsyncProgress *progress,
     eos_updater_set_downloaded_bytes (updater, bytes);
 }
 
-static gboolean
-repo_pull (OstreeRepo *self,
-           const gchar *remote_name,
-           const gchar *ref,
-           const gchar *url_override,
-           OstreeAsyncProgress *progress,
-           GCancellable *cancellable,
-           GError **error)
+static GVariant *
+get_options_for_pull (const gchar *ref,
+                      const gchar *url_override,
+                      gboolean     disable_static_deltas)
 {
   g_auto(GVariantBuilder) builder = { { { 0, } } };
-  g_autoptr(GVariant) options = NULL;
 
   g_variant_builder_init (&builder, G_VARIANT_TYPE ("a{sv}"));
   g_variant_builder_add (&builder, "{s@v}", "refs",
                          g_variant_new_variant (g_variant_new_strv (&ref, 1)));
   if (url_override != NULL)
     g_variant_builder_add (&builder, "{s@v}", "override-url",
                            g_variant_new_variant (g_variant_new_string (url_override)));
+  g_variant_builder_add (&builder, "{s@v}", "disable-static-deltas",
+                         g_variant_new_variant (g_variant_new_boolean (disable_static_deltas)));
 
-  options = g_variant_ref_sink (g_variant_builder_end (&builder));
-  return ostree_repo_pull_with_options (self, remote_name, options,
-                                        progress, cancellable, error);
+  return g_variant_ref_sink (g_variant_builder_end (&builder));
```

Comment:
Yeah, I'm always sinking variants, as I don't like the idea of floating refs. But will change the code.